test/test_stepsize.py

```diff
@@ -47,6 +47,31 @@ def test_adaptive_stepsize_shrinks(cutback_factor, target):
     expected_value = current_value * my_stepsize.cutback_factor
     assert np.isclose(new_value, expected_value)
 
+@pytest.mark.parametrize("max_stepsize, growth_factor, target", [(4,3,1)])
+def test_max_stepsize(max_stepsize, growth_factor, target):
+    """Checks that the stepsize is capped at max
+    stepsize.
+
+    Args:
+        max_stepsize (float): maximum stepsize
+        growth_factor (float): the growth factor
+        target (int): the target number of iterations
+    """
+
+    my_stepsize = F.Stepsize(initial_value=2)
+    my_stepsize.max_stepsize = max_stepsize
+    my_stepsize.growth_factor = growth_factor
+    my_stepsize.target_nb_iterations = target
+
+    current_value = 2
+    new_value = my_stepsize.modify_value(
+        value=current_value,
+        nb_iterations=my_stepsize.target_nb_iterations - 1,
+    )
+
+    expected_value = max_stepsize
+    assert new_value == expected_value
```

Comment:
Can you try and adapt this test to catch the behaviour I mentioned above, or write another test?